addons/mail/models/mail_message.py

```diff
@@ -902,7 +902,12 @@ def _record_by_message(self):
             message: self.env[message.model]
             .browse(message.res_id)
             .with_prefetch(records_by_model_name[message.model]._prefetch_ids)
-            for message in self.filtered(lambda m: m.model and m.res_id)
+            for message in self.filtered(
```

Comment:
Feels like it should be done before calling this helper. Its usage is to classify messages, not filter them. Isn't it possible to do it in _to_store instead ? It can filter out unwanted message, then consider they are ok.

Reply:
So if I follow this pattern of "earliest in the flow", I guess we should filter them out in message fetch instead.